### PR TITLE
Fix broken migration

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
The existing version of the migration replaced null values with
empty arrays. The presence of null values is used by the queries
to discriminate create and exercise events. This caused an
illegal state exception when reading active contracts for the API
(which are represented by their create events on the events table)
as the reader was mistakenly reading exercise events and create
events.

changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by gerolf-da — APPROVED** [review]
Thank you, @stefanobaghino-da!

File: ledger/participant-integration-api/src/main/scala/db/migration/postgres/V38__Update_value_versions.scala [hunk 1]
```diff
@@ -73,18 +71,17 @@ private[migration] final class V38__Update_value_versions extends BaseJavaMigrat
...
-        NamedParameter(
-          label,
-          ValueSerializer
-            .serializeValue(
-              value.copy(stableValueVersion),
-              s"failed to serialize $label for $tableName $rowKey"))
-      case (label, None) =>
-        NamedParameter(label, EmptyArray)
-    }
+  ): NamedParameter = {
```
Comment (SamirTalwar-DA) [inline ledger/participant-integration-api/src/main/scala/db/migration/postgres/V38__Update_value_versions.scala:74]: Perhaps if this returned an `Option` and it was converted into a `NamedParameter` in a separate operation, the logic would be clearer.

Alternatively you could add an explicit type to `newValue`.

What do you think?

**Review by SamirTalwar-DA — COMMENTED** [review]
Good catch. How have you tested it?

File: ledger/participant-integration-api/src/main/scala/db/migration/postgres/V38__Update_value_versions.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/participant-integration-api/src/main/scala/db/migration/postgres/V38__Update_value_versions.scala:74]: My initial idea was to re-write the thing to make it dumber, but I'd prefer changing this as little as possible.

**Comment by stefanobaghino-da** [comment]
> Good catch. How have you tested it?

@SamirTalwar-DA Manual testing with in-house application. Inspected the database, ensured that functionality of the app. This error condition only pops up when using "very old" pre-1.0 ledgers and the compatibility tests cannot cover this, unfortunately.

**Comment by SamirTalwar-DA** [comment]
Sounds good, thanks!

**Pull request merged**